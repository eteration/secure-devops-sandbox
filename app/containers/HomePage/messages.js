/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.HomePage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Secure DevOps!',
  },
  detail: {
    id: `${scope}.detail`,
    defaultMessage:
      'Cloud & Kubernetes provides a rapid, iterative development experience for teams. Azure can provide an end-to-end secure devops solution for your application with minimal effort.',
  },
});
