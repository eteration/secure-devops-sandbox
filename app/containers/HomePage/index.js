/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { createSelector } from 'reselect';

import { changeLocale } from '../LanguageProvider/actions';
import { makeSelectLocale } from '../LanguageProvider/selectors';
import messages from './messages';

function HomePage(props) {
  return (
    <div style={{ textAlign: 'center' }}>
      <h1>
        <FormattedMessage {...messages.header} />
      </h1>
      <p>
        <FormattedMessage {...messages.detail} />
      </p>
      <div style={{ textAlign: 'center', background: 'black', padding: 20 }}>
        <button type="button" onClick={() => props.onLocaleToggle('en')}>
          en
        </button>{' '}
        -{' '}
        <button type="button" onClick={() => props.onLocaleToggle('tr')}>
          tr
        </button>
        <br />
        <br />
        <img
          alt="Meetups Eteration"
          src="https://meetups.eteration.com/logo.png"
        />
        <p style={{ color: 'white' }}>meetups.eteration.com</p>
      </div>
    </div>
  );
}

HomePage.propTypes = {
  onLocaleToggle: PropTypes.func,
  locale: PropTypes.string,
};

const mapStateToProps = createSelector(
  makeSelectLocale(),
  locale => ({
    locale,
  }),
);

export function mapDispatchToProps(dispatch) {
  return {
    onLocaleToggle: lang => dispatch(changeLocale(lang)),
    dispatch,
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(HomePage);
