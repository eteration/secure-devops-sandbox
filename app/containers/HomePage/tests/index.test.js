import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import configureStore from '../../../configureStore';

import HomePage from '../index';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
