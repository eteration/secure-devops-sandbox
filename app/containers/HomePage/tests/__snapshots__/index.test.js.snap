// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div
  style="text-align: center;"
>
  <h1>
    <span>
      Secure DevOps!
    </span>
  </h1>
  <p>
    <span>
      Cloud & Kubernetes provides a rapid, iterative development experience for teams. Azure can provide an end-to-end secure devops solution for your application with minimal effort.
    </span>
  </p>
  <div
    style="text-align: center; background: black; padding: 20px;"
  >
    <button
      type="button"
    >
      en
    </button>
     
    -
     
    <button
      type="button"
    >
      tr
    </button>
    <br />
    <br />
    <img
      alt="Meetups Eteration"
      src="https://meetups.eteration.com/logo.png"
    />
    <p
      style="color: white;"
    >
      meetups.eteration.com
    </p>
  </div>
</div>
`;
